ncm-iptables/src/main/perl/iptables.pm

```diff
@@ -407,113 +426,105 @@ sub GetPathEntries {
-		$rule->{$key} = $opresult;
-	    }
-	}
-
-	if ( defined $options_arg{$key} && $options_arg{$key} ne "" ) {
-	    $aux = $options_arg{$key};
-	    if ($rule->{$key} !~ /^$aux$/ && $key =~ /^$val$/){
-		my $skip = 0;
-		foreach (@{$iptables_totality{$table}{targets}}) {
-		    $skip = 1 if $_ eq $rule->{$key};
-		}
-		next if $skip;
-		push(@{$iptables_totality{$table}{targets}}, $rule->{$key});
-		$iptables_totality{$table}{user_targets}{$rule->{$key}} = 1;
-	    }
-	}
-    }
-
-    if (defined $rule->{'-j'}) {
-	$target = $rule->{'-j'};
-	$target =~ tr/A-Z/a-z/;
-    }
-
 
-    if ( defined $entries->{$table}->{ordered_rules} &&
-	 $entries->{$table}->{ordered_rules} eq "yes" ) {
-	$target = "ordered";
-    }
+    $self->debug(5, "Entering method GetResource");
 
-    if ( defined $cnt->{$target} ) {
-	next if ( ! &find_rule($rule,$entries->{$table}->{rules}->{$target}->{$cnt->{$target}}) );
-	$entries->{$table}->{rules}->{$target}->{$cnt->{$target}} = $rule;
-	$cnt->{$target}++;
-    }
+    my $entries = $self->GetPathEntries( $path, $config );
+    return $entries if $?;
 
-}
+    foreach my $table (keys %iptables_totality) {
+        next if (!defined $entries->{$table});
+
+        #define the regular expressions for -N, -A etc based on the specific targets for table
+        my $tmp = $self->uppercase($self->regExp(@{$iptables_totality{$table}{chains}}));
+        $OPTION_VALIDATORS{$_} = $tmp foreach (@{$iptables_totality{$table}{commands}});
+        $OPTION_VALIDATORS{'-j'} = $self->uppercase($self->regExp(@{$iptables_totality{$table}{targets}}));
+
+        $entries->{$table} = $self->GetPathEntries("$path/$table", $config);
+        next if $?;
+
+        $entries->{$table}->{preamble} = $self->GetPathEntries("$path/$table/preamble", $config);
+
+        my $cnt = {};
+        foreach my $target (@{$iptables_totality{$table}{targets}}) {
+            $cnt->{$target} = 0;
+        }
+
+        $entries->{$table}->{rules} = $self->GetPathEntries("$path/$table/rules", $config);
+        next if $?;
+
+        RULE: foreach my $name (sort { $a <=> $b } keys %{$entries->{$table}->{rules}}) {
+            next if ($name !~ /^\d+$/);
+            my $rule = $self->GetPathEntries( "$path/$table/rules/$name", $config );
+            return if $?;
+            $self->rule_options_translate($rule);
+
+            if (!defined $rule->{chain}) {
+                $? = 7;
+                $@ = "missed chain entry on rule \"$path/$table/rules/$name\"";
+                return {};
+            }
+
+            if (defined $rule->{-j}) {
+                #check if exists
+                if ($self->uppercase($rule->{-j}) !~ /$OPTION_VALIDATORS{'-j'}/) {
+                    $iptables_totality{$table}{user_targets}{$self->uppercase($rule->{-j})} = 1;
+                }
+            }
+
+            $rule->{command}=$iptables_totality{$table}{commands}[0] if(!defined $rule->{command} || $rule->{command} eq "");
```

Comment:
spaces around `=`; if `$rule->{command}` cannot be `0`, use `if (! $rule->{command})`